lib/chai/core/assertions.js

```diff
@@ -988,9 +988,33 @@ module.exports = function (chai, _) {
 
   function assertInstanceOf (constructor, msg) {
     if (msg) flag(this, 'message', msg);
+
+    var target = flag(this, 'object')
+    var validInstanceOfTarget = constructor === Object(constructor) && (
+        typeof constructor === 'function' ||
+        (typeof Symbol !== 'undefined' && typeof Symbol.hasInstance !== 'undefined' &&
+         typeof constructor[Symbol.hasInstance] === 'function')
+    );
+
+    if (!validInstanceOfTarget) {
+      var constructorType = constructor === null ? 'null' : typeof constructor;
+      throw new Error('The instanceof assertion needs a constructor but ' + constructorType + ' was given.');
+    }
+
+    var isInstanceOf;
+    try {
+        isInstanceOf = target instanceof constructor
+    } catch (err) {
+        throw new Error(target + ' instanceof ' + constructor + ' failed: ' + err.message + '.')
```

Comment:
As @meeber noticed, I also think we're going a bit full-circle here. The whole purpose of this PR was to give our users more friendly error messages, but given what was said I'm not sure this is the best choice anymore.

So, what do you think? Should we cancel the `Error` related change and keep the other improvements or should we keep it?

I'd like to hear more opinions about that.